### PR TITLE
Copr: Add `DecimalDivide` to `RPN` expression framework

### PR DESCRIPTION
**Pull request by iosmanthus**
## What have you changed? (mandatory)
Migrates `DivideDecimal` to `DecimalDivide`

## What are the type of changes? (mandatory)
- New feature (a change which adds functionality)

## How has this PR been tested? (mandatory)
`cargo test impl_arithmetic`

### CODE REVIEW

**Comment by mapleFU** [comment]
Looks nice.

File: src/coprocessor/dag/rpn_expr/impl_arithmetic.rs [hunk 1]
```diff
@@ -1144,4 +1165,18 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn test_decimal_divide() {
+        let test_cases = vec![("2.2", "1.1", "2.0")];
```
Comment (breeswish) [inline src/coprocessor/dag/rpn_expr/impl_arithmetic.rs]: insufficient test cases:

```
            (
                ScalarFuncSig::DivideDecimal,
                str2dec("12.3"),
                str2dec("-0.3"),
                str2dec("-41"),
            ),
            (
                ScalarFuncSig::DivideDecimal,
                str2dec("12.3"),
                str2dec("0.3"),
                str2dec("41"),
            ),
            (
                ScalarFuncSig::DivideDecimal,
                str2dec("12.3"),
                str2dec("0"),
                Datum::Null,
            ),
            (
                ScalarFuncSig::DivideDecimal,
                Datum::Null,
                str2dec("123"),
                Datum::Null,
            ),
            (
                ScalarFuncSig::DivideDecimal,
                str2dec("123"),
                Datum::Null,
                Datum::Null,
            ),

            (
                ScalarFuncSig::DivideDecimal,
                str2dec("12.3"),
                str2dec("0.0"),
            ),
            (
                ScalarFuncSig::DivideDecimal,
                str2dec("12.3"),
                str2dec("-0.0"),
            ),
```

File: src/coprocessor/dag/rpn_expr/impl_arithmetic.rs [hunk 2]
```diff
@@ -1144,4 +1165,30 @@ mod tests {
...
+    #[test]
+    fn test_decimal_divide() {
+        let cases = vec![
+            (str2dec("2.2"), str2dec("1.1"), str2dec("2.0")),
+            (str2dec("12.3"), str2dec("-0.3"), str2dec("-41")),
+            (str2dec("12.3"), str2dec("0.3"), str2dec("41")),
+            (str2dec("12.3"), str2dec("0"), None),
+            (None, str2dec("2"), None),
+            (str2dec("123"), None, None),
+        ];
```
Comment (mapleFU) [inline src/coprocessor/dag/rpn_expr/impl_arithmetic.rs]:
```
            (
                ScalarFuncSig::DivideDecimal,
                str2dec("12.3"),
                str2dec("0.0"),
            ),
            (
                ScalarFuncSig::DivideDecimal,
                str2dec("12.3"),
                str2dec("-0.0"),
            ),
```
I notice there are another 2 tests in builtin_arithmetic.rs 195-211 lines
Would you mind and these tests here?

Comment (iosmanthus) [reply src/coprocessor/dag/rpn_expr/impl_arithmetic.rs]: I covered these cases by:
```rust
(str2dec("12.3"), str2dec("0"), None),
```

**Comment by breeswish** [comment]
In the original test case, the following two also checks whether errors are correctly generated when flag `SqlMode::ERROR_FOR_DIVISION_BY_ZERO` is set:

```
            (
                ScalarFuncSig::DivideDecimal,
                str2dec("12.3"),
                str2dec("0.0"),
            ),
            (
                ScalarFuncSig::DivideDecimal,
                str2dec("12.3"),
                str2dec("-0.0"),
            ),
```

**Pull request merged**